### PR TITLE
Go templates for the provided.al2 runtime

### PR DESCRIPTION
**Pull request by bmoffatt**
*Issue #, if available:*

https://github.com/aws/aws-sam-cli-app-templates/issues/181

*Description of changes:*

This mostly is a copy+paste of the go1.x templates, with changes to adapt them for use with the `provided.al2` runtime.

Changes to the various template.yml look like:
``` diff
 Resources:
   HelloWorldFunction:
     Type: AWS::Serverless::Function # More info about Function Resource: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction
+    Metadata:
+      BuildMethod: go1.x
     Properties:
       CodeUri: hello-world/
-      Handler: hello-world
-      Runtime: go1.x
+      Handler: bootstrap
+      Runtime: provided.al2
       {%- if cookiecutter.architectures.value != []%}
       Architectures:
       {%- for arch in cookiecutter.architectures.value %}
```

And changes too the cookiecutter.json look like:
``` diff
 {
     "project_name": "Name of the project",
-    "runtime": "go1.x",
+    "runtime": "provided.al2",
     "architectures": {
-        "value": []
+        "value": [
+            [ "x86_64", "arm64" ]
+        ]
     },
     "_copy_without_render": [
         ".gitignore"
     ]
```

I'm not sure if this change to `cookiecutter.json` is sufficient to add arm support to the sam init prompts for these templates. So let me know if there's additional changes I need to make to wire it up.

I also copy+pasted the test cases I could find, and verified before submitting this PR by configuring CodeBuild batch builds in my AWS account.  If there are any test cases I missed, or any manual verification I should do, let me know! 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: manifest-v2.json [hunk 1]
```diff
@@ -204,6 +204,41 @@
       "useCaseName": "Multi-step workflow"
     }
   ],
+  "go (provided.al2)": [
+    {
+      "directory": "go1.x/cookiecutter-aws-sam-hello-golang",
```
Comment (lucashuy) [inline manifest-v2.json:209]: Just an update to the path (for all the new al2 golang templates)

```suggestion
      "directory": "provided.al2/go/cookiecutter-aws-sam-hello-golang",
```

**Review by lucashuy — CHANGES_REQUESTED** [review]
Hi, thanks for opening this PR to add go templates with the AL2 runtime! This PR looks good to me so far, just have a minor change to make

File: manifest-v2.json (hunk 1 above)
Comment (bmoffatt) [reply manifest-v2.json:209]: Thanks for catching! I thought I'd done this edit 🤦‍♂️

**Review by lucashuy — APPROVED** [review]
Thanks for making those changes, this looks good to me!

**Pull request merged**